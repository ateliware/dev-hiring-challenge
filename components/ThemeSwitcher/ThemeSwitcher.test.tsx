import { fireEvent, render, screen } from "@testing-library/react";
import { AvailableThemes, ThemeSwitcher } from ".";
import "../Footer/matchMedia.mock";

describe("ThemeSwitcher", () => {
  const switcherId = "themeSwitcherTest";
  it("renderiza o seletor de temas", () => {
    const wrapper = render(<ThemeSwitcher switcherId={switcherId} />);
    const select = screen.getByTestId(
      "themeSwitcherSelect"
    ) as HTMLSelectElement;
    const label = screen.getByText("tema", { selector: "span" });
    expect(label).toBeInTheDocument();
    expect(label).toHaveAttribute("id", switcherId);
    expect(select).toHaveAttribute("aria-labelledby", switcherId);
    fireEvent.change(select, { target: { value: AvailableThemes.Light } });
    expect(select.value).toBe(AvailableThemes.Light);
    fireEvent.change(select, { target: { value: AvailableThemes.Dark } });
    expect(select.value).toBe(AvailableThemes.Dark);
    //TODO criar um teste para verificar que o atributo "theme" do elemento HTML mude quando o valor do select Ã© mudado
    expect(wrapper.container).toMatchSnapshot();
  });
});
