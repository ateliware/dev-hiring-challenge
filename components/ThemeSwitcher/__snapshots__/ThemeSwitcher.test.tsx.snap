// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitcher renderiza o seletor de temas 1`] = `
<div>
  <div
    class="main"
  >
    <span
      class="label"
      id="themeSwitcherTest"
    >
      tema
    </span>
    <hr
      class="divider"
    />
    <select
      aria-labelledby="themeSwitcherTest"
      class="select"
      data-testid="themeSwitcherSelect"
      name="tema"
      placeholder="tema"
      title="tema"
    >
      <option
        value="light"
      >
        claro
      </option>
      <option
        value="dark"
      >
        escuro
      </option>
    </select>
  </div>
</div>
`;
